src/partition/growfs.c

```diff
@@ -0,0 +1,91 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "device-nodes.h"
+#include "escape.h"
+#include "fd-util.h"
+#include "format-util.h"
+#include "log.h"
+#include "missing.h"
+#include "mount-util.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "strv.h"
+
+int main(int argc, char *argv[]) {
+        dev_t devno;
+        _cleanup_close_ int mountfd = -1, devfd = -1;
+        size_t size;
+        int blocksize;
+        uint64_t numblocks;
+        char devpath[DEV_NUM_PATH_MAX], fb[FORMAT_BYTES_MAX];
+        int r;
+
+        if (argc != 2) {
+                log_error("This program requires one argument (the mountpoint).");
+                return EXIT_FAILURE;
+        }
+
+        log_set_target(LOG_TARGET_AUTO);
+        log_parse_environment();
+        log_open();
+
+        r = path_is_mount_point(argv[1], NULL, 0);
+        if (r < 0) {
+                log_error_errno(r, "Failed to check if \"%s\" is a mount point: %m", argv[1]);
+                return EXIT_FAILURE;
+        }
+        if (r == 0) {
+                log_error_errno(r, "\"%s\" is not a mount point: %m", argv[1]);
+                return EXIT_FAILURE;
+        }
+
+        r = get_block_device(argv[1], &devno);
+        if (r < 0) {
+                log_error_errno(r, "Failed to determine block device of \"%s\": %m", argv[1]);
+                return EXIT_FAILURE;
+        }
+
+        mountfd = open(argv[1], O_RDONLY);
```

Comment:
please always add O_CLOEXEC, we never know what might be forked off in the background by some library...